pages/wiki/openembedded.md

```diff
@@ -3,10 +3,245 @@ title: OpenEmbedded
+84dcc94da3adb52b53ae4fa38fe49e5d,Unknown
+NOTE: Starting bitbake server...
+INFO: Recipe /home/developer/asteroid/build/workspace/recipes/hello/hello_git.bb has been automatically created; further editing may be required to make it fully functional
+```
+
+So the tool has created a recipe based on just the name we have given and the directory containing the source.  Since it seems unsure about the license and suggests that further editing might be required, let's take it's advice and edit the recipe.  The simplest way to do that is again to use `devtool`:
+
+```
+devtool edit-recipe hello
+```
+
+This will open the recipe in an editor.  It says that the license is unknown, so we change the line from `LICENSE = "Unknown"` to `LICENSE = "GPL-3.0-only"`.  Not that the `SRC_URI` is set to point to a git repository.  This is because `devtool` is smart enough to figure out that even though you gave it a local directory, it discovered that the source was actually git and made the appropriate changes.  We can now attempt to build.
+
+```
+devtool build hello
+```
+
+However this fails because we're still missing something.  In this case, we're missing two things.  First, instead of the `inherit cmake` line, it should be `inherit cmake_qt5` and second, we need to add the following line just below that:
+
+```
+DEPENDS += "qml-asteroid asteroid-generate-desktop-native qttools-native qtdeclarative-native"
+```
+
+With those two small changes, we should now get a successful build.  
+
+As before, we can easily test on the watch:
+
+```
+devtool deploy-target hello root@192.168.2.15
+```
+
+This will automatically install all of the install targets from the `hello` project onto the watch.  Start the launcher on the watch and scroll to the end of your list of apps, and you should now see a "HELLO WORLD APP" at the end.  Running it will show "Hello World!" on the screen.
+
+Removing it from the watch is also the same as before:
+
+```
+devtool undeploy-target hello root@192.168.2.15
+```
+
+
+# Sharing your changes
+
+If you make a change to AsteroidOS software that you think might be useful to others, such as a bug fix or feature enhancement, there are several ways to share it, depending on what exactly is being changed.   
+
+ - creating a recipe patch file
+ - creating a pull request for an AsteroidOS project
+ - creating an installable package (`.ipk`)
+
+## Creating a recipe patch file
+
+Many of the software packages that make up an AsteroidOS image are not unique to AsteroidOS.  They may be general purpose tools such as `systemctl` or watch-specific software from other open source repositories such as `mcetool`.  For these, there may be a need to introduce a patch just for AsteroidOS.  To do so, first, within the appropriate `workspace` directory where you made changes, use `git commit` to commit the changes to the local repository.  Once this is done, automatically create the patch.  For our example, we could do that with this command:
```

Comment:
One also needs to stage the files using `git add`.
It might be better to remove the `git` related parts and just mention that someone needs to commit the changes using git and link to a good git tutorial if need be (@FlorentRevest had one in the past, but I'm not sure where to find it anymore).

Reply:
I'm assuming developers know how to develop, but it's not unreasonable to provide a link to some help.  Can you suggest a link?

Reply:
I added a link to an atlassian git tutorial.